lib/metadata/schema-parser.js

```diff
@@ -1009,37 +1009,24 @@ function getTokenToReplicaSimpleMapper(replicationFactor) {
  * @private
  */
 function getTokenToReplicaNetworkMapper(replicationFactors) {
-  //                A(1)
+  //                A(DC1)
   //
-  //           H         B(2)
+  //           H         B(DC2)
   //                |
-  //      G       --+--       C(1)
+  //      G       --+--       C(DC1)
   //                |
-  //           F         D(2)
+  //           F         D(DC2)
   //
-  //                E(1)
+  //                E(DC1)
   return (function tokenNetworkStrategy(tokenizer, ring, primaryReplicas, datacenters) {
-    function isDoneForToken(replicasByDc) {
-      for (var dc in replicationFactors) {
-        if (!replicationFactors.hasOwnProperty(dc)) {
-          continue;
-        }
-        var rf = Math.min(parseInt(replicationFactors[dc], 10), datacenters[dc]);
-        if (replicasByDc[dc] < rf) {
-          return false;
-        }
-      }
-      return true;
-    }
-    //For each token
-    //Get an Array of tokens
-    //Checking that there aren't more tokens per dc than specified by the replication factors
     var replicas = {};
     for (var i = 0; i < ring.length; i++) {
       var token = ring[i];
       var key = tokenizer.stringify(token);
       var tokenReplicas = [];
```

Comment:
If you are using vnodes, a replica can have multiple tokens.  In the case where not enough replicas are found to meet CL, it's possible tokenToReplicas will be populated with the same replicas multiple times.  In this case, we should skip repeated replicas and move on to the next replica.